src/sema/mutability.rs

```diff
@@ -252,6 +252,18 @@ fn read_expression(expr: &Expression, state: &mut StateCheck) -> bool {
         Expression::Subscript { loc, array_ty, .. } if array_ty.is_contract_storage() => {
             state.read(loc)
         }
+        Expression::Builtin {
+            kind: Builtin::FunctionSelector,
```

Comment:
Is this binding necessary?

Reply:
I am not following.  Here we are filtering on `kind == Builtin::FunctionSelector`

Reply:
Of course, I was not paying attention